api/events/types.go

```diff
@@ -129,14 +129,30 @@ const (
 
 type Range struct {
 	Unit RangeType
-	From uint64
-	To   uint64
+	From *uint64 `json:"from,omitempty"`
+	To   *uint64 `json:"to,omitempty"`
+}
+
+func NewRange(unit RangeType, from uint64, to uint64) *Range {
```

Comment:
Seems test only?

Reply:
Yepp, its just annoying to do this everytime:

```golang
to := 0
from := 1

range := &Range{
  from: &from
  to: &to
}
```

Reply:
How about put the func to test file?